rust/rsc/src/rsc/main.rs

```diff
@@ -44,30 +49,51 @@ struct ServerOptions {
 
     #[arg(help = "Launches the cache without an external database", long)]
     standalone: bool,
+
+    #[arg(
+        help = "Specify an absolute path for a local directory based store",
+        value_name = "LOCAL_STORE",
+        long
+    )]
+    local_store: Option<String>,
 }
 
-fn create_router(state: Arc<DatabaseConnection>) -> Router {
+fn create_router(conn: Arc<DatabaseConnection>, config: Arc<config::RSCConfig>) -> Router {
     Router::new()
         .route(
             "/job",
             post({
-                let shared_state = state.clone();
-                move |body| add_job::add_job(body, shared_state)
+                let conn = conn.clone();
+                move |body| add_job::add_job(body, conn)
             })
             .layer(axum::middleware::from_fn({
-                let shared_state = state.clone();
-                move |req, next| {
-                    api_key_check::api_key_check_middleware(req, next, shared_state.clone())
-                }
+                let conn = conn.clone();
+                move |req, next| api_key_check::api_key_check_middleware(req, next, conn.clone())
             })),
         )
         .route(
             "/job/matching",
             post({
-                let shared_state = state.clone();
-                move |body| read_job::read_job(body, shared_state)
+                let conn = conn.clone();
+                move |body| read_job::read_job(body, conn)
+            }),
+        )
+        .route(
+            "/blob",
+            get({
+                let config = config.clone();
+                move || blob::get_upload_url(config.server_addr.clone())
             }),
         )
+        .route(
+            "/blob",
+            post({
+                let conn = conn.clone();
+                let config = config.clone();
```

Comment:
You don't need to pass the config here, just the blob store

Reply:
@V-FEXrt This is my only blocking comment and its not a hard blocker. I just want use to construct a *single* blob store that gets passed to multiple routes on demand (using an Arc and cloning)

Reply:
How should I return an http error when the configuration is wrong?

Reply:
nvm I have a solution I'm happy with